node/src/test_utils.rs

```diff
@@ -57,3 +70,94 @@ pub fn test_node_config_with_keypair(keypair: Keypair) -> NodeConfig<InMemorySto
         ..test_node_config()
     }
 }
+
```

Comment:
I would also keep a `impl P2p { ... }` related to mocking there, not in the p2p.rs

Reply:
I didn't want to make `pub(crate)` the fields of `P2p`

Reply:
we could do the `pub(crate::test_utils)` but it isn't anyway nicer :thinking: then leaving that up to you